### ISSUE TITLE
schema-reader Error: Method ExtendedBlueprint::multiLineString does not exist.

### ISSUE THREAD

**Issue by LeonardoPessatti** [VemtoOrg/vemto2-issues#16]
**Error Message:**
schema-reader Error: Method ExtendedBlueprint::multiLineString does not exist.

**Stack:**
```
#0 C:\wamp64\www\epicbrew\database\migrations\2024_03_13_083809_001_create_universe_table.php(19): Illuminate\Database\Schema\Blueprint->__call('multiLineString', Array)
#1 phar://C:/Program Files/Vemto/resources/static/VMTTL1/classes/ExtendedBuilder.php(21): Illuminate\Database\Migrations\Migration@anonymous->{closure}(Object(ExtendedBlueprint))
#2 C:\wamp64\www\epicbrew\vendor\laravel\framework\src\Illuminate\Support\helpers.php(345): ExtendedBuilder->{closure}(Object(ExtendedBlueprint))
#3 phar://C:/Program Files/Vemto/resources/static/VMTTL1/classes/ExtendedBuilder.php(18): tap(Object(ExtendedBlueprint), Object(Closure))
#4 C:\wamp64\www\epicbrew\vendor\laravel\framework\src\Illuminate\Support\Facades\Facade.php(357): ExtendedBuilder->create('universe', Object(Closure))
#5 C:\wamp64\www\epicbrew\database\migrations\2024_03_13_083809_001_create_universe_table.php(15): Illuminate\Support\Facades\Facade::__callStatic('create', Array)
#6 [internal function]: Illuminate\Database\Migrations\Migration@anonymous->up()
#7 phar://C:/Program Files/Vemto/resources/static/VMTTL1/classes/MigrationDecoder.php(22): ReflectionMethod->invoke(Object(Illuminate\Database\Migrations\Migration@anonymous))
#8 phar://C:/Program Files/Vemto/resources/static/VMTTL1/index.php(77): MigrationDecoder->decode()
#9 phar://C:/Program Files/Vemto/resources/static/VMTTL1/common/Vemto.php(107): {closure}()
#10 phar://C:/Program Files/Vemto/resources/static/VMTTL1/index.php(18): Vemto::execute('schema-reader', Object(Closure))
#11 C:\Program Files\Vemto\resources\static\VMTTL1(14): require('phar://C:/Progr...')
#12 {main}
```

---

Vemto 2.0.3
win32 10.0.22621
Locale: pt-BR
App version: 2.0.3

**Comment by TiagoSilvaPereira** [comment]
@LeonardoPessatti is this a new Laravel project or an existing one?

**Comment by LeonardoPessatti** [comment]
It is a fresh install.
Just added Laravel, Jetstream, created a table using Vemto and tried to run.

**Comment by TiagoSilvaPereira** [comment]
Got it. Could you please try the new version of 2.0.4?

**Comment by LeonardoPessatti** [comment]
- I tried, now throws the same error when trying to connect the project folder. 
  - Even after removing the .vemto folder.
  - Even after reinstalling 2.0.4 (Also, a few visual bugs disappeared after reinstalling 👍).
- Also, i tried clonning the project (containing only the fresh install) in another folder and throwed another error, I'm pasting it below. 
  - I can share the Github project with you, so you can have a look if it may help.
  - PHP is added to PATH and "php" command runs without problems on cmd.

**(error) FAILED to execute command: "php" "C:\ProgramFiles\Vemto\resources\static\VMTTL1"**

`Warning: require(C:\wamp64\www\Epic-Brew/vendor/autoload.php): Failed to open stream: No such file or directory in phar://C:/Program Files/Vemto/resources/static/VMTTL1/load.php on line 7
Call Stack:
    0.0002     584960   1. {main}() C:\Program Files\Vemto\resources\static\VMTTL1:0
    0.0568     945832   2. require('phar://C:/Program Files/Vemto/resources/static/VMTTL1/index.php') C:\Program Files\Vemto\resources\static\VMTTL1:14
    0.0570     947136   3. require_once('phar://C:/Program Files/Vemto/resources/static/VMTTL1/load.php') phar://C:/Program Files/Vemto/resources/static/VMTTL1/index.php:4
Fatal error: Uncaught Error: Failed opening required 'C:\wamp64\www\Epic-Brew/vendor/autoload.php' (include_path='.;C:\php\pear') in phar://C:/Program Files/Vemto/resources/static/VMTTL1/load.php on line 7
Error: Failed opening required 'C:\wamp64\www\Epic-Brew/vendor/autoload.php' (include_path='.;C:\php\pear') in phar://C:/Program Files/Vemto/resources/static/VMTTL1/load.php on line 7
Call Stack:
    0.0002     584960   1. {main}() C:\Program Files\Vemto\resources\static\VMTTL1:0
    0.0568     945832   2. require('phar://C:/Program Files/Vemto/resources/static/VMTTL1/index.php') C:\Program Files\Vemto\resources\static\VMTTL1:14
    0.0570     947136   3. require_once('phar://C:/Program Files/Vemto/resources/static/VMTTL1/load.php') phar://C:/Program Files/Vemto/resources/static/VMTTL1/index.php:4
schema-reader Error: Method ExtendedBlueprint::multiLineString does not exist.
schema-reader Error: #0 C:\wamp64\www\epicbrew\database\migrations\2024_03_13_083809_001_create_universe_table.php(19): Illuminate\Database\Schema\Blueprint->__call('multiLineString', Array)
#1 phar://C:/Program Files/Vemto/resources/static/VMTTL1/classes/ExtendedBuilder.php(21): Illuminate\Database\Migrations\Migration@anonymous->{closure}(Object(ExtendedBlueprint))
#2 C:\wamp64\www\epicbrew\vendor\laravel\framework\src\Illuminate\Support\helpers.php(345): ExtendedBuilder->{closure}(Object(ExtendedBlueprint))
#3 phar://C:/Program Files/Vemto/resources/static/VMTTL1/classes/ExtendedBuilder.php(18): tap(Object(ExtendedBlueprint), Object(Closure))
#4 C:\wamp64\www\epicbrew\vendor\laravel\framework\src\Illuminate\Support\Facades\Facade.php(357): ExtendedBuilder->create('universe', Object(Closure))
#5 C:\wamp64\www\epicbrew\database\migrations\2024_03_13_083809_001_create_universe_table.php(15): Illuminate\Support\Facades\Facade::__callStatic('create', Array)
#6 [internal function]: Illuminate\Database\Migrations\Migration@anonymous->up()
#7 phar://C:/Program Files/Vemto/resources/static/VMTTL1/classes/MigrationDecoder.php(22): ReflectionMethod->invoke(Object(Illuminate\Database\Migrations\Migration@anonymous))
#8 phar://C:/Program Files/Vemto/resources/static/VMTTL1/index.php(77): MigrationDecoder->decode()
#9 phar://C:/Program Files/Vemto/resources/static/VMTTL1/common/Vemto.php(107): {closure}()
#10 phar://C:/Program Files/Vemto/resources/static/VMTTL1/index.php(18): Vemto::execute('schema-reader', Object(Closure))
#11 C:\Program Files\Vemto\resources\static\VMTTL1(14): require('phar://C:/Progr...')
#12 {main}
schema-reader Error: Method ExtendedBlueprint::multiLineString does not exist.
schema-reader Error: #0 C:\wamp64\www\epicbrew\database\migrations\2024_03_13_083809_001_create_universe_table.php(19): Illuminate\Database\Schema\Blueprint->__call('multiLineString', Array)
#1 phar://C:/Program Files/Vemto/resources/static/VMTTL1/classes/ExtendedBuilder.php(21): Illuminate\Database\Migrations\Migration@anonymous->{closure}(Object(ExtendedBlueprint))
#2 C:\wamp64\www\epicbrew\vendor\laravel\framework\src\Illuminate\Support\helpers.php(345): ExtendedBuilder->{closure}(Object(ExtendedBlueprint))
#3 phar://C:/Program Files/Vemto/resources/static/VMTTL1/classes/ExtendedBuilder.php(18): tap(Object(ExtendedBlueprint), Object(Closure))
#4 C:\wamp64\www\epicbrew\vendor\laravel\framework\src\Illuminate\Support\Facades\Facade.php(357): ExtendedBuilder->create('universe', Object(Closure))
#5 C:\wamp64\www\epicbrew\database\migrations\2024_03_13_083809_001_create_universe_table.php(15): Illuminate\Support\Facades\Facade::__callStatic('create', Array)
#6 [internal function]: Illuminate\Database\Migrations\Migration@anonymous->up()
#7 phar://C:/Program Files/Vemto/resources/static/VMTTL1/classes/MigrationDecoder.php(22): ReflectionMethod->invoke(Object(Illuminate\Database\Migrations\Migration@anonymous))
#8 phar://C:/Program Files/Vemto/resources/static/VMTTL1/index.php(77): MigrationDecoder->decode()
#9 phar://C:/Program Files/Vemto/resources/static/VMTTL1/common/Vemto.php(107): {closure}()
#10 phar://C:/Program Files/Vemto/resources/static/VMTTL1/index.php(18): Vemto::execute('schema-reader', Object(Closure))
#11 C:\Program Files\Vemto\resources\static\VMTTL1(14): require('phar://C:/Progr...')
#12 {main}`

**Comment by TiagoSilvaPereira** [comment]
Hi @LeonardoPessatti I think I know what is happening. Apparently, your project has a version of Laravel that does not yet support the *multiLineString* method on migrations Blueprint. What Laravel version is this project using?

I still need to improve the error messages to provide better clues as to what is going on. For example, I need to instruct the user to update the Laravel version when necessary, etc.

In the case of the second error, after cloning the project, did you run ```composer install```? 

Seems Vemto is unable to find the PHP files to read. I need to put a better message in this case, too.

**Comment by LeonardoPessatti** [comment]
> What Laravel version is this project using?

The project is running Laravel 11.
![image](https://github.com/TiagoSilvaPereira/vemto2-issues/assets/11570844/7082fa8c-36c7-4a7f-952c-fce57640fbf6)

After I changed the multiLineString to String on the migration, the problem is gone.
Before closing Vemto and being unable to reopen the folder, i changed the column type on Vemto, but it didn't changed in the migration, as the migration had something wrong. Maybe letting the user make changes even when the code is triggering errors beforehand. 

> In the case of the second error, after cloning the project, did you run composer install?

Yes, i did.

**Comment by TiagoSilvaPereira** [comment]
@LeonardoPessatti thank you for the info. I'm studying do change the schema reader to a simpler architecture which will avoid most of the problems with booting a Laravel application (and it should be faster too)

**Comment by TiagoSilvaPereira** [comment]
@LeonardoPessatti Sorry for the late reply. I just had time to take a deeper look. The problem is that **multiLineString** was discontinued on Laravel 11. I'll remove this type of column from Laravel 11 projects

**Comment by TiagoSilvaPereira** [comment]
Fixed on 2.0.7
